apps/passport-client/components/screens/LoginScreens/OneClickLoginScreen.tsx

```diff
@@ -0,0 +1,51 @@
+import { useCallback, useEffect } from "react";
+import { useDispatch, useQuery, useSelf } from "../../../src/appHooks";
+import { MaybeModal } from "../../modals/Modal";
+import { AppContainer } from "../../shared/AppContainer";
+import { ScreenLoader } from "../../shared/ScreenLoader";
+
+export function OneClickLoginScreen(): JSX.Element | null {
+  const dispatch = useDispatch();
+  const query = useQuery();
+  const email = query?.get("email");
+  const code = query?.get("code");
+  const targetFolder = query?.get("targetFolder");
+
+  const self = useSelf();
+
+  const handleOneClickLogin = useCallback(async () => {
+    if (!email || !code) {
```

Comment:
```suggestion
    if (!email || !code || self) {
```
maybe?

Reply:
if self is defined, it actually goes into the other useEffect below and redirects. but i'm actually realizing that these three check need to be on the same level